Add group attributes to documentation

mekaneck opened the pull request:
Added missing documentation for the attributes of group entities.

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
The documentation for group entities did not cover what attributes were populated for a group entity. This fixes that. 


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit


- **New Features**
	- Enhanced documentation with new sections detailing attributes of new-style and old-style groups.
	- Added tables outlining key attributes for better comprehension of group behavior.

- **Documentation**
	- Clarified differences in behavior between group types, aiding users in managing configurations effectively.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

mekaneck committed:
Update group.markdown with group attributes

Added missing documentation for the attributes of group entities.

mekaneck committed:
Correction to the old-style group attributes behavior

old-style groups will become `unknown` when all members are `unavailable`.

mekaneck committed:
Merge branch 'current' into current

frenck commented on source/_integrations/group.markdown:
```diff
@@ -155,6 +155,14 @@ To add or remove entities from an existing group, click on `Group options`, all
 
+### Group attributes
+
+These are the attributes available for a group. Note that when the group's members are all `unavailable` the group itself becomes `unavailable` and all attributes are removed.
```
The latter seems a bug? As in, why would we remove the `entity_id` attribute when the group becomes unavailable?

mekaneck replied:
I agree it is undesirable, but my understanding is that every entity (group or any other domain) loses all their attributes when they become unavailable. The desired change would therefore be to make the entity go `unknown` instead. But I expect that to be a feature request and not an issue. There is a current issue on this (https://github.com/home-assistant/core/issues/124055) but previous ones have gone stale. Hence my expectation it needs to be a FR. And therefore we should document the current behavior until this is completed.

frenck replied:
> And therefore we should document the current behavior until this is completed.

I think it is a bug, we shouldn't be documenting bugs. We should fix them.

mekaneck replied:
I've removed the comments related to this behavior.

mekaneck committed:
Update group.markdown to eliminate reference to bug

Removed comment regarding attributes disappearing when the group becomes `unknown`. Also cleaned up the attribute definitions in the table for the old-style group.

mekaneck committed:
Merge branch 'current' into current

frenck reviewed (approved):
Thanks, @mekaneck 👍 

../Frenck
